libgalois/include/katana/Properties.h

```diff
@@ -505,5 +505,112 @@ struct StructProperty
   }
 };
 
+/// This is a reference to a node's instance of ArrayProperty
+///
+/// \tparam T the C type of array elements.
+template <typename T>
+class ArrayRef {
+public:
+  using value_type = T;
+  using iterator = value_type*;
+  using const_iterator = const value_type*;
+
+  ArrayRef(T* ptr, size_t size) : data_(ptr), size_(size) {}
+
+  T& operator[](size_t index) { return data_[index]; }
+  const T& operator[](size_t index) const { return data_[index]; }
+  size_t size() { return size_; }
+
+  T* begin() { return iterator(&data_[0]); }
+  const T* begin() const { return const_iterator(&data_[0]); }
+  T* end() { return iterator(&data_[size_]); }
+  const T* end() const { return const_iterator(&data_[size_]); }
+
+private:
+  T* data_;
+  size_t size_;
+};
+
+/// ArrayPropertyView provides a property view over arrow::LargeListArrays of arrow:Arrays
+/// of elements with trivial constructors (std::is_trivial) and standard layout
+/// (std::is_standard_layout). (i.e., POD)
+///
+/// \tparam T the C type of array elements.
+/// Currently, only double is supported. Support for other types need to be added.
+/// \tparam N the size of the arrays in arrow::LargeLIstArray
+template <typename T, size_t N>
+class ArrayPropertyView {
+public:
+  using value_type = ArrayRef<const T>;
+  using reference = ArrayRef<T>;
+  using const_reference = const ArrayRef<T>;
```

Comment:
I believe const_reference should also be `ArrayRef<const T>` (the current const has no effect, as the functions below return a const_reference by value, but using `const T` as the template parameter ensures that if `ArrayPropertyView` is const, the `ArrayRef` it returns contains const data, even if the ArrayRef itself isn't const)